substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaUtilZipSubstitutions.java

```diff
@@ -383,75 +504,146 @@ private int inflateBytes(long addr, byte[] b, int off, int len) throws DataForma
             } else if (ret == ZLib.Z_MEM_ERROR()) {
                 throw new OutOfMemoryError();
             } else {
-                throw new InternalError();
+                throw new InternalError(CTypeConversion.toJavaString(strm.msg()));
+            }
+        }
+    }
+
+    // (JDK 11)src/java.base/share/native/libzip/Inflater.c-static jlong doInflate(JNIEnv *env, jobject this, jlong addr,
+    //                       jbyte *input, jint inputLen,
+    //                       jbyte *output, jint outputLen)
+    private long doInflate(long addr, CCharPointer input, int inputLen, CCharPointer output, int outputLen) throws DataFormatException {
```

Comment:
This method needs to move to `Util_java_util_zip_Inflate`, since it is not a method in the target class for which you are writing substitutions.

(Did you try building with both JDK-8 and JDK-11?  You should have gotten errors from the build on JDK-8.  I haven't tried the JDK-11 build yet.)

Reply:
Do you want me to make these re-factorings to get the code to build for JDK-8?  Or would you like that challenge?